File: app/src/main/java/com/photovault/worker/BootReceiver.java
package com.photovault.worker;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import androidx.work.Constraints;
import androidx.work.ExistingWorkPolicy;
import androidx.work.NetworkType;
import androidx.work.OneTimeWorkRequest;
import androidx.work.WorkManager;

/**
 * Restarts the upload worker after device reboot.
 */
public class BootReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            scheduleUploadWorker(context);
        }
    }

    static void scheduleUploadWorker(Context context) {
        var prefs = context.getSharedPreferences("photovault_settings", Context.MODE_PRIVATE);
        boolean wifiOnly = prefs.getBoolean("wifi_only", true);

        Constraints constraints = new Constraints.Builder()
                .setRequiredNetworkType(wifiOnly ? NetworkType.UNMETERED : NetworkType.CONNECTED)
                .setRequiresBatteryNotLow(true)
                .build();

        OneTimeWorkRequest request = new OneTimeWorkRequest.Builder(UploadWorker.class)
                .setConstraints(constraints)
                .build();

        WorkManager.getInstance(context)
                .enqueueUniqueWork("upload_queue",
                        ExistingWorkPolicy.KEEP, request);
    }
}

File: app/src/main/java/com/photovault/service/BootReceiver.kt
package com.photovault.service

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import androidx.work.ExistingWorkPolicy
import androidx.work.OneTimeWorkRequestBuilder
import androidx.work.WorkManager
import com.photovault.service.worker.UploadWorker

class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == Intent.ACTION_BOOT_COMPLETED) {
            val request = OneTimeWorkRequestBuilder<UploadWorker>().build()
            WorkManager.getInstance(context).enqueueUniqueWork(
                "resume_uploads",
                ExistingWorkPolicy.KEEP,
                request
            )
        }
    }
}
